lib/max_subarray.js

```diff
@@ -1,9 +1,25 @@
 
-// Time Complexity:
-// Space Complexity:
+// Time Complexity: O(n) n is the size of the array
+// Space Complexity: O(1)
```

Comment:
✨ Notice how better time complexity this approach achieves over a "naïve" approach of checking for the maximum achievable sum starting from every position and every length. The correctness of this approach might not be apparent, so I definitely encourage reading a bit more about it. [This](https://towardsdatascience.com/maximum-subarray-problem-and-kadanes-algorithm-4cb1ce91be72#8dc2) has a fairly good explanation, as well as a description of why this is considered a dynamic programming approach (on the face it might not "feel" like one).

Since like the fibonacci sequence, we are able to maintain a sliding window of recent values to complete our calculation, we can do it with a constant O(1) amount of storage.